src/services/pcn-router/src/Router.cpp

```diff
@@ -775,6 +775,75 @@ void Router::addPorts(const std::string &name, const PortsJsonObject &conf) {
   add_port<PortsJsonObject>(name, conf);
   if (get_shadow())
     getPorts(name)->setMac(getPorts(name)->getMac());
+
+  // Creates function objects containing the callback functions
+  // to be called when there are notifications on one netdev
+  std::function<void(const std::string, const std::string)> f_ip = std::bind(&Router::iface_ip_notification,
+                                this, std::placeholders::_1, std::placeholders::_2);
+  std::function<void(const std::string, const std::string)> f_mac = std::bind(&Router::iface_mac_notification,
+                                this, std::placeholders::_1, std::placeholders::_2);
+
+  // Register the new port to IP and MAC notifications arriving from netdevs
+  get_port(name)->subscribe_event("IP_ADDRESS", f_ip);
```

Comment:
This looks a little bit weird to me, why to handle all this logic at the Router level?, This is related to ports, so all of this should be moved to the Ports class.